src/core/load-fragment.c

```diff
@@ -5630,7 +5630,7 @@ int config_parse_bpf_foreign_program(
         r = extract_first_word(&rvalue, &word, ":", 0);
         if (r == -ENOMEM)
                 return log_oom();
-        if (r < 0) {
+        if (r <= 0) {
```

Comment:
Could you also update `log_error_errno()` -> `log_syntax()` and return `0` after `cgroup_add_bpf_foreign_program() below.`

Reply:
Updated, except that `cgroup_add_bpf_foreign_program()` can only fails with -ENOMEM, so I left it as it was.

Reply:
Ah, I see. Thanks.